### PR TITLE
feat: Create chat page and configure chat route

### LINKED ISSUE #15: Create chat page

### PR DESCRIPTION
**Pull request by BaoNguyen09**
## Changes made
- chat page styles: Add style to messages and polls. Make most components generally responsive (textbox will size up for larger text, screen will auto-scroll for new messages, etc.). I also updated the styles of poll form popup
- chat page feature: enable live-updating messages when chat load initially as well as receiving chat updates when using
- chat controller: Add pin validation for `/chat/:chat_id` route and redirect user to error page with custom error message if not valid. I also switched to chat pin instead of chat id and modified all tests

### CODE REVIEW

File: backend/controllers/__test__/chatController.test.js [hunk 1]
```diff
@@ -257,15 +258,11 @@ describe("chatController", () => {
...
-        title: "Test Chat",
-        messages: [],
-        polls: [],
-        users: [user],
-      });
-      const invalid_timestamp = "invalid-date";
-      await load_chat(chat._id, invalid_timestamp);
+        const user = await UserService.create_guest_user();
+        const pin = await generate_unique_pin();
+        const chat = await Chat.create({ title: "Test Chat", pin: pin, messages: [], polls: [], users: [user]});
```
Comment (matthewsong18) [inline backend/controllers/__test__/chatController.test.js]: How does this line work? 
```javascript
.create({ title: "Test Chat", pin: pin, message: [], polls: [], users: [user]});
```
`users` should expect an ObjectId, so it should be `users: [user._id]` here, no? Not sure how this didn't error.

**Review by matthewsong18 — CHANGES_REQUESTED** [review]
The tests pass, but I'm not sure about  your Chat.create calls

File: backend/controllers/__test__/chatController.test.js (hunk 1 above)
Comment (BaoNguyen09) [reply backend/controllers/__test__/chatController.test.js]: yeah you're right

**Review by matthewsong18 — CHANGES_REQUESTED** [review]
Bao, there are multiple `users: [user]` calls in your code. I suggest running a find for it and fixing them all.

**Review by matthewsong18 — DISMISSED** [review]
We can fix the issues later

**Review by matthewsong18 — APPROVED** [review]

**Pull request merged**